components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java

```diff
@@ -206,17 +250,176 @@ public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
         context = mock(PrivilegedCarbonContext.class);
         when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(context);
 
-        RegistryService registry = mock(RegistryService.class);
+        registry = mock(RegistryService.class);
         mockStatic(OAuth2ServiceComponentHolder.class);
         when(OAuth2ServiceComponentHolder.getRegistryService()).thenReturn(registry);
 
-        UserRegistry userRegistry = mock(UserRegistry.class);
+        userRegistry = mock(UserRegistry.class);
         when(registry.getConfigSystemRegistry(SAMPLE_TENANT_ID)).thenReturn(userRegistry);
 
-        Resource resource = mock(Resource.class);
+        resource = mock(Resource.class);
         when(userRegistry.get(OAuthConstants.SCOPE_RESOURCE_PATH)).thenReturn(resource);
 
         samlAssertionClaimsCallback.handleCustomClaims(jwtClaimsSet, oAuthAuthzReqMessageContext);
         assertTrue(jwtClaimsSet.getAllClaims().isEmpty());
     }
+
+    @Test
+    public void testCustomClaimForOAuthTokenReqMessageContextWithNullAssertionSubject() throws Exception {
+        jwtClaimsSet = mock(JWTClaimsSet.class);
```

Comment:
Can't we use actual objects for JWTClaimsSet, Subject,  NameID and Attribute? If we are mocking these objects, need to mock their functionalities as well.